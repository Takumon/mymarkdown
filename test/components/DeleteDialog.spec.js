import { mount, shallowMount, createLocalVue } from '@vue/test-utils'
import Vuex, { mapState, mapActions } from 'vuex'
import VueMaterial from 'vue-material'
import TestTargetVue from '../../components/DeleteDialog.vue'


jest.useFakeTimers();

describe('DeleteDialog.vue', () => {

  let wrapper
  let store
  let deleteMemoSpy

  afterEach(() => {
    deleteMemoSpy.mockReset()
    deleteMemoSpy.mockRestore()
  })

  beforeEach(() => {
    const localVue = createLocalVue()

    localVue.use(VueMaterial)
    localVue.use(Vuex)

    deleteMemoSpy = jest.fn(() => Promise.resolve(true))

    store = new Vuex.Store({
      state: {
        showDeletingDialog: true,
      },
      actions: {
        setShowDeletingDialog ( { commit }, isShow) {
          return new Promise((resolve, reject) => {
            commit( 'setShowDeletingDialog', { isShow })
            resolve()
          })
        },
      },
      mutations: {
        setShowDeletingDialog (state, payload) {
          state.showDeletingDialog = payload.isShow
        },
      }
    })

    wrapper = mount(TestTargetVue, {
      store,
      localVue,
      mixins: [{
        methods: {
          deleteMemo: deleteMemoSpy
        }
      }],
      attachToDocument: true
    })
    jest.runAllTimers()
  })

  describe('初期表示', () => {

    test('スナップショット', () => {
      const $dialog = wrapper.find('[data-test="delete-dialog"]')
      expect($dialog).toMatchSnapshot()
    })

    test('ダイアログが表示される', () => {
      expect(store.state.showDeletingDialog).toBe(true)
      const $dialogTitle = wrapper.find('[data-test="delete-dialog"] .md-title')
      expect($dialogTitle.isVisible()).toBe(true)
    })
  })

  describe('キャンセルボタンクリック時', () => {
    beforeEach(() => {
      const $cancelButton = wrapper.findAll('[data-test="delete-dialog"] button').at(0)
      expect($cancelButton.text()).toBe('Cancel')
      $cancelButton.trigger('click')
      jest.runAllTimers()
    })

    test('ダイアログが非表示', () => {
      store.dispatch('setShowDeletingDialog', false)
    })
  })

  describe('削除ボタンクリック時', () => {
    beforeEach(() => {
      expect(deleteMemoSpy).toHaveBeenCalledTimes(0)
      const $cancelButton = wrapper.findAll('[data-test="delete-dialog"] button').at(1)
      expect($cancelButton.text()).toBe('Delete')
      $cancelButton.trigger('click')
      jest.runAllTimers()
    })

    test('削除処理が呼ばれている', () => {
      expect(deleteMemoSpy).toHaveBeenCalledTimes(1)
    })

    test('ダイアログが非表示', () => {
      expect(store.state.showDeletingDialog).toBe(false)
    })

  })

})
