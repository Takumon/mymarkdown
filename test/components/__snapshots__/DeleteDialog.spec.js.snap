// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteDialog.vue 初期表示 スナップショット 1`] = `
<div class="md-dialog md-theme-default" data-test="delete-dialog" md-active="true">
  <div class="md-dialog-container" tabindex="-1"><span class="md-dialog-title md-title">Delete the memo?</span>
    <div class="md-dialog-content">Once deleted, it can not be restored.
      <br>Do you really want to delete this?</div>
    <div class="md-dialog-actions">
      <button type="button" class="md-button md-theme-default">
        <div class="md-ripple">
          <div class="md-button-content">Cancel</div>
        </div>
      </button>
      <button type="button" class="md-button md-primary md-theme-default">
        <div class="md-ripple">
          <div class="md-button-content">Delete</div>
        </div>
      </button>
    </div>
  </div>
</div>
`;
